api/lib/infrastructure/repositories/user-repository.js

```diff
@@ -465,15 +465,15 @@ function _toMembershipsDomain(membershipsBookshelf) {
   });
 }
 
-function _toPixAdminRolesDomain(pixAdminRolesBookshelf) {
-  return pixAdminRolesBookshelf.map((pixAdminRoleBookshelf) => {
-    return new PixAdminRole({
-      id: pixAdminRoleBookshelf.get('id'),
-      role: pixAdminRoleBookshelf.get('role'),
-      userId: pixAdminRoleBookshelf.get('userId'),
-      createdAt: pixAdminRoleBookshelf.get('createdAt'),
-      updatedAt: pixAdminRoleBookshelf.get('updatedAt'),
-      disabledAt: pixAdminRoleBookshelf.get('disabledAt'),
+function _toAdminMemberDomain(adminMembersBookshelf) {
```

Comment:
Question  : est-ce qu'on a vraiment besoin de lui passer un objet en entier ? 
Dans le modèle user on utiliser actuellement
  ```
  get hasRoleSuperAdmin() {
    return !!this.pixAdminRoles.find(({ role, disabledAt }) => role === ROLES.SUPER_ADMIN && !disabledAt);
  } 
  ```
 Vous voyez d'autres utilisations ?

Reply:
Ce code actuel est un reliquat du premier ticket de migration de la base de données. Comme tu l'indiques, le besoin de retourner un tableau d'objets de rôles n'est plus nécessaire, de même que passer un objet complet.

Ce sera l'objet d'un ticket TECH pour faire cette modification car il y a un risque que la taille de cette PR soit plus importante.

Proposition:
- api
  - retirer le tableau pour passer d'une relation One-to-Many à One-to-One
- front.s
  - idem que pour l'api avec les fonctionnalités des models Ember

Reply:
> * En faisant l'essai d'une modification de mon propre rôle j'ai fait planté l'appli (c'est rigolo comme dit Anne G) mais on devrait avoir quelque chose de plus propre que ça. 
> *

C'est corrigé dans cette PR https://github.com/1024pix/pix/pull/4361/commits car on ne peut pas se connecter actuellement avec un autre rôle que `SUPER_ADMIN`

Cf mon message de ce matin dans slack, il faudrait remplacer `checkUserHasRoleSuperAdmin` par `checkAuthorizationToAccesAdminScope` pour qu'un utilisateur autre que super admin puisse naviguer sur pix admin sans problème. Pas testé encore le nombre de fichiers à modifier et les effets de bords sur les autres fichiers (test api) et sur les tests du front